extensions/position-size-calculator/src/utils.ts

```diff
@@ -0,0 +1,164 @@
+        }
+        if (typeof fixedRiskAmt === "undefined" || isNaN(fixedRiskAmt) || fixedRiskAmt <= 0) {
+          return { error: "Fixed Risk Amount is invalid or missing" };
+        }
+        // RRR is needed for target calculation in this mode as per our design
+        if (typeof rrr === "undefined" || isNaN(rrr) || rrr <= 0) {
+          return { error: "Reward/Risk Ratio is invalid or missing for target calculation" };
+        }
+        if (entry <= slPrice) {
+          return { error: "Entry must be above Stop Loss" };
+        }
+        sl = slPrice;
+        target = entry + (entry - sl) * rrr;
+        break;
+
+      default: {
+        // Should not happen if mode is correctly typed
+        const exhaustiveCheck: never = mode;
+        return { error: `Unknown calculation mode: ${exhaustiveCheck}` };
+      }
+    }
+
+    // --- Core Calculations ---
+    const riskPerShare = entry - sl;
+    if (riskPerShare <= 0) {
+      return {
+        error: "Risk per share is zero or negative. Check Entry and Stop Loss.",
+      };
+    }
+
+    if (mode === "Fixed Risk") {
+      // We've already checked fixedRiskAmt is defined and valid for this mode
+      riskAmount = fixedRiskAmt as number; // Safe assertion after check
+      if (riskAmount > capital) {
+        return { error: "Fixed Risk Amount exceeds Capital" };
+      }
+    } else {
+      riskAmount = capital * (riskPercent / 100);
+    }
+
+    const positionSize = riskAmount / riskPerShare;
+    const investmentAmount = positionSize * entry;
+
+    // Ensure target is valid after all calculations
+    if (target <= entry) {
+      return { error: "Calculated Target is not above Entry. Review inputs." };
+    }
+
+    const reward = (target - entry) * positionSize;
+    // RRR can be recalculated or taken from input if mode is RRR-Based
+    const rewardRiskRatio = (target - entry) / riskPerShare;
+    const stopLossPercentCalc = ((sl - entry) / entry) * 100;
+    const targetPercentCalc = ((target - entry) / entry) * 100;
+    const exposurePercent = (investmentAmount / capital) * 100;
+
+    return {
+      mode,
+      capital,
+      riskAmount,
+      entry,
+      stopLoss: sl,
+      target,
+      riskPerShare,
+      positionSize: Math.floor(positionSize),
+      investmentAmount,
+      reward,
+      rewardRiskRatio,
+      stopLossPercent: stopLossPercentCalc,
+      targetPercent: targetPercentCalc,
+      exposurePercent,
+    };
+  } catch (e) {
+    console.error("Calculation Error:", e);
+    return { error: "An unexpected error occurred during calculation." };
+  }
+}
+
+// Helper to format currency
+export const formatCurrency = (value: number): string => {
+  if (isNaN(value)) return "N/A";
+  return value.toLocaleString("en-IN", {
+    // Example: Indian Rupee format
+    style: "currency",
+    currency: "INR",
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  });
+};
```

Comment:
Can we make this extension generic instead of using INR? It could be used by everyone!